File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0], maximumProfit = 0;
        
        for (int price : prices) {
            min = Math.min(min, price);
            maximumProfit = Math.max(maximumProfit, price - min);
        }
        
        return maximumProfit;
    }
}

File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var min = Int.MAX_VALUE
        var maxProfit = 0

        for (price in prices) {
            min = minOf(min, price)
            maxProfit = maxOf(maxProfit, price - min)
        }

        return maxProfit
    }
}